library/math/conversions.lisp

```diff
@@ -182,5 +182,32 @@ cannot be represented in :TO. These fall into a few categories:
                  (Err "Integer to Double-Float conversion out-of-range")
                  (Ok y)))))))
 
+(cl:defmacro integer-tryinto-float (integer integer-name lisp-float float float-name pow)
+  `(coalton-toplevel
+     (define-instance (TryInto ,integer ,float)
+       (define (tryInto x)
+         (lisp (Result String ,float) (x)
+	   (cl:if (cl:and (cl:> x (cl:expt -2 ,pow)) (cl:< x (cl:expt 2 ,pow)))	 
+	          (cl:let ((y (cl:ignore-errors (cl:coerce x (cl:quote ,lisp-float)))))
```

Comment:
Don't ignore errors here. The conversion should never fail, and if it does we want to know.